python/paddle/fluid/backward.py

```diff
@@ -1209,9 +1209,8 @@ def append_backward(loss,
     is_in_control_flow = current_block_idx != 0
 
     # Double grad is not supported in sub-block (control flow)
-    if not is_in_control_flow:
-        # _appending_grad_times used for double grad
-        program._appending_grad_times += 1
+    # Note(zhiqiu): This should be refined. 
+    program._appending_grad_times += 1
```

Comment:
If we always invoke `prune_backward` even with a forward network, I think that this changing can be removed. In fact, we need to remove `_appending_grad_times`  from `Program` because: (1) it is  a bit hard coding; (2) this attribute cannot be serialized and deserialized.

Reply:
Agreed, `_appending_grad_times` is not a good design.